compiler/src/symboltable.rs

```diff
@@ -273,9 +299,9 @@ impl SymbolTableBuilder {
         Ok(symbol_table)
     }
 
-    fn enter_block(&mut self, name: &str) {
+    fn enter_block(&mut self, name: &str, typ: SymbolTableType, line_number: usize) {
```

Comment:
Is enter_block correctly named? It's job is to create a new symbol table, but blocks don't create scopes in python. Should this be enter_scope, or push_table or something?

Reply:
`enter_scope` would be a better name this function. I'll change it later on, I want to refactor this code more.